src/Components/Components/src/Routing/RouteTableFactory.cs

```diff
@@ -19,29 +19,31 @@ namespace Microsoft.AspNetCore.Components;
 /// <summary>
 /// Resolves components for an application.
 /// </summary>
-internal static class RouteTableFactory
+internal class RouteTableFactory
 {
-    private static readonly ConcurrentDictionary<RouteKey, RouteTable> Cache = new();
+    public static readonly RouteTableFactory Instance = new();
```

Comment:
My concern with statics is that they won't work correctly in a multi-tenant scenario. If that was keyed by something like a `Type` I would assume it's fine but here it's a `RouteKey` and I don't know if there could be collisions between tenants (whatever they would mean). If `RouteTableFactory` was itself a singleton in DI that would mitigate that as one can assume that a tenant would have a custom DI container, or decide to share the DI container singletons. With a static there is no opportunity to change the way it's shared.

Reply:
The Blazor router is not DI enabled unfortunately, so we are not making this scenario any better/worse than it was.

Reply:
We will have to make some changes in 9.0 to make the router in Blazor more flexible.

Reply:
Actually, it shouldn't matter, since we pass in the list of assemblies to the router, it should work in a multi-tenancy context too.